app/models/user.rb

```diff
@@ -29,4 +35,16 @@ def self.from_omniauth(auth)
       user.avatar_url = auth.info.image
     end
   end
+
+  # Validation for email: symbol . (dot) provided that it is neither the first nor the last,
+  # and also if it is not repeated more than once in a row.
+  def email_dots
+    login = email.scan(/\S+@/).join
+    errors.add(:email, 'has too many dots') if login.count('.') > 1 || login[0] == '.' || login[-2] == '.'
```

Comment:
For some emails it is ok to be like 

first_name.last_name@site.com

But this is a question of the requirements